templates/workspaces/base/terraform/azure-monitor/azure-monitor.tf

```
@@ -0,0 +1,106 @@
+resource "azurerm_log_analytics_workspace" "core" {
```

```suggestion
resource "azurerm_log_analytics_workspace" "workspace" {
```

Plus, creating a new log analytics workspace requires also setting up its customer provisioned storage account, doesn't it?
Something like this:
```terraform
resource "azurerm_log_analytics_linked_storage_account" "workspace_storage_ingestion" {
  data_source_type      = "Ingestion"
  resource_group_name   = var.resource_group_name
  workspace_resource_id = azurerm_log_analytics_workspace.core.id
  storage_account_ids   = [azurerm_storage_account.logs.id]
}

resource "azurerm_log_analytics_linked_storage_account" "workspace_storage_customlogs" {
  data_source_type      = "customlogs"
  resource_group_name   = var.resource_group_name
  workspace_resource_id = azurerm_log_analytics_workspace.core.id
  storage_account_ids   = [azurerm_storage_account.logs.id]
}
```

---

Added - thanks